pkg/generate/app/sourcelookup.go

```diff
@@ -15,6 +16,11 @@ import (
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
+var (
+	argumentGit         = regexp.MustCompile("^(http://|https://|git@|git://).*(?:#([a-zA-Z0-9]*))?$")
+	argumentGitProtocol = regexp.MustCompile("^(git@|git://)")
```

Comment:
@rhcarvalho not offhand, sorry.  i'm just suggesting you strip it for the validation check, leave it on otherwise.